ciao-controller/compute.go

```diff
@@ -906,7 +911,16 @@ func createServer(w http.ResponseWriter, r *http.Request, context *controller) {
 	}
 	servers.TotalServers = len(instances)
 
-	b, err := json.Marshal(servers)
+	// set machine ID for OpenStack compatibility
+	server.Server.ID = instances[0].ID
```

Comment:
Yes, I think that should do it.  I'd change the MinInstances check as well and maybe the check  https://github.com/01org/ciao/blob/master/ciao-controller/command.go#L127 as well.